exchanges/okx/okx.go

```diff
@@ -4290,42 +4292,73 @@ func (ok *Okx) SystemStatusResponse(ctx context.Context, state string) ([]System
 }
 
 // GetAssetTypeFromInstrumentType returns an asset Item instance given and Instrument Type string.
-func GetAssetTypeFromInstrumentType(instrumentType string) (asset.Item, error) {
+func GetAssetTypeFromInstrumentType(instrumentType string) asset.Item {
 	switch strings.ToUpper(instrumentType) {
 	case okxInstTypeSwap, okxInstTypeContract:
-		return asset.PerpetualSwap, nil
-	case okxInstTypeANY:
-		return asset.Empty, nil
-	default:
-		return asset.New(instrumentType)
+		return asset.PerpetualSwap
+	case okxInstTypeSpot:
+		return asset.Spot
+	case okxInstTypeMargin:
+		return asset.Margin
+	case okxInstTypeFutures:
+		return asset.Futures
+	case okxInstTypeOption:
+		return asset.Options
+	}
+	return asset.Empty
+}
+
+// GetAssetsFromInstrumentTypeOrID parses an instrument type and instrument ID and returns a list of assets
+// that the currency pair is associated with.
+func (ok *Okx) GetAssetsFromInstrumentTypeOrID(instType, instrumentID string) ([]asset.Item, error) {
+	if instType != "" {
+		a := GetAssetTypeFromInstrumentType(instType)
+		if a != asset.Empty {
+			return []asset.Item{a}, nil
+		}
 	}
-}
-
-// GuessAssetTypeFromInstrumentID returns or guesses the instrument id.
-func (ok *Okx) GuessAssetTypeFromInstrumentID(instrumentID string) (asset.Item, error) {
-	if strings.HasSuffix(instrumentID, okxInstTypeSwap) {
-		return asset.PerpetualSwap, nil
+	if instrumentID == "" {
+		return nil, fmt.Errorf("%w instrumentID", errEmptyArgument)
+	}
+	splitSymbol := strings.Split(instrumentID, ok.CurrencyPairs.RequestFormat.Delimiter)
+	if len(splitSymbol) <= 1 {
+		return nil, fmt.Errorf("%w %v", currency.ErrCurrencyNotSupported, instrumentID)
+	}
+	pair, err := currency.NewPairDelimiter(instrumentID, ok.CurrencyPairs.RequestFormat.Delimiter)
+	if err != nil {
+		return nil, err
 	}
-	count := strings.Count(instrumentID, currency.DashDelimiter)
 	switch {
-	case count >= 3:
-		return asset.Options, nil
-	case count == 2:
-		return asset.Futures, nil
-	default:
-		pair, err := currency.NewPairFromString(instrumentID)
-		if err != nil {
-			return asset.Empty, err
+	case len(splitSymbol) == 2:
+		resp := make([]asset.Item, 0, 2)
+		if err := ok.CurrencyPairs.IsAssetPairEnabled(asset.Spot, pair); err == nil {
+			resp = append(resp, asset.Spot)
 		}
-		pairs, err := ok.GetEnabledPairs(asset.Margin)
-		if err != nil {
-			return asset.Empty, err
+		if err := ok.CurrencyPairs.IsAssetPairEnabled(asset.Margin, pair); err == nil {
+			resp = append(resp, asset.Margin)
+		}
+		if len(resp) > 0 {
+			return resp, nil
 		}
-		for x := range pairs {
-			if pairs[x].Equal(pair) {
-				return asset.Margin, nil
+	case len(splitSymbol) > 2:
+		resp := make([]asset.Item, 1)
+		switch splitSymbol[len(splitSymbol)-1] {
+		case "SWAP", "swap":
+			if err := ok.CurrencyPairs.IsAssetPairEnabled(asset.PerpetualSwap, pair); err == nil {
+				resp[0] = asset.PerpetualSwap
+				return resp, nil
```

Comment:
```go 
return []asset.Item{asset.PerpetualSwap}
```

and below

Reply:
Only reason why I didn't was because I defined the slice above as having a limit of 1.... Though I never actually tested if it did anything better/different 👀